hw3-sp24/src/routes/customRoute.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef, useState } from "react";
+import NavBar from "../components/navbar";
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+} from "chart.js";
+import "chart.js/auto";
+import { Doughnut } from "react-chartjs-2";
+import axios from "axios";
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+);
+
+ChartJS.defaults.color = "#000000";
+ChartJS.defaults.borderColor = "#000000";
+
+const countryAPIURL: string =
+  "https://cs464p564-frontend-api.vercel.app/api/countries";
+
+interface Country {
+  languages: string;
+}
+
+export default function CustomRoute() {
+  const hasFetched = useRef(false);
+  const currentLoc = window.location.href;
+  const [countryData, setCountryData] = useState<Array<Country>>([]);
+
+  useEffect(() => {
+    if (!hasFetched.current) {
+      hasFetched.current = true;
+      axios
+        .get(countryAPIURL)
+        .then(response => {
+          const data: Array<Country> = response.data.map(
+            (item: { official_languages: string }) => ({
+              languages: item.official_languages,
+            }),
+          );
+          setCountryData(data);
+        })
+        .catch(error => console.error(error.message));
+    }
+  });
+
+  const labels = Array.from(
+    new Set(countryData.flatMap(country => country.languages)),
+  );
+
+  const languageCountMap = new Map<string, number>();
+  countryData.forEach(country => {
+    const languages = Array.isArray(country.languages)
+      ? country.languages
+      : [country.languages];
+    languages.forEach(language => {
+      languageCountMap.set(language, (languageCountMap.get(language) || 0) + 1);
+    });
+  });
+  const languageCounts: number[] = Array.from(languageCountMap.values());
+
+  const data = {
+    labels,
+    datasets: [
+      {
+        label: "Spoken Langauges Totals In South America",
+        data: languageCounts,
+        backgroundColor: [

```

Comment:
Maybe consider making the colors a separate file like Caterina did in utils and then import it here.